{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.0 Find Run Numbers for ChIP-seq data from this paper:\n",
    "CD8+ T Cells Utilize Highly Dynamic Enhancer Repertoires and Regulatory Circuitry in Response to Infections"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1.1) Search the GEO accension number from the paper. The Accension number is GSE76031."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1.2) The SubSeries for ChIP-seq data is GSE76029."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1.3) Search for ChIP-seq data accession number in SRA Run Selector."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1.4) Save RunInfo Table"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.0 Build a folder named Sample_fastq ###"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "2.1) Redirect to the new folder."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /home/rtu/\n",
    "mkdir Sample_fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Applied-Bioinformatics\n",
      "Configuration.md\n",
      "hisat2-2.1.0\n",
      "hisat_out\n",
      "Homo_sapiens.GRCh38.94.gtf\n",
      "Homo_sapiens.GRCh38.94.txt\n",
      "miniconda3\n",
      "Miniconda3-latest-Linux-x86_64.sh\n",
      "ncbi\n",
      "R\n",
      "README.md\n",
      "Sample_fastq\n",
      "sratoolkit.2.9.2-ubuntu64\n",
      "sratoolkit.current-ubuntu64.tar.gz\n",
      "SRR5454079_1.fastq\n",
      "SRR5454080_1.fastq\n",
      "SRR5830640_1.fastq\n",
      "SRR5830640_2.fastq\n",
      "test-repo\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/rtu/\n",
    "ls\n",
    "echo\n",
    "cd /home/rtu/Sample_fastq/\n",
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "2.2) Save the first 500 spots of each file into Sample_fastq directory with a for loop."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read 500 spots for SRR3001750\n",
      "Written 500 spots for SRR3001750\n",
      "Read 500 spots for SRR3001751\n",
      "Written 500 spots for SRR3001751\n",
      "Read 500 spots for SRR3001752\n",
      "Written 500 spots for SRR3001752\n",
      "Read 500 spots for SRR3001753\n",
      "Written 500 spots for SRR3001753\n",
      "Read 500 spots for SRR3001754\n",
      "Written 500 spots for SRR3001754\n",
      "Read 500 spots for SRR3001755\n",
      "Written 500 spots for SRR3001755\n",
      "Read 500 spots for SRR3001756\n",
      "Written 500 spots for SRR3001756\n",
      "Read 500 spots for SRR3001757\n",
      "Written 500 spots for SRR3001757\n",
      "Read 500 spots for SRR3001758\n",
      "Written 500 spots for SRR3001758\n",
      "Read 500 spots for SRR3001759\n",
      "Written 500 spots for SRR3001759\n",
      "Read 500 spots for SRR3001760\n",
      "Written 500 spots for SRR3001760\n",
      "Read 500 spots for SRR3001761\n",
      "Written 500 spots for SRR3001761\n",
      "Read 500 spots for SRR3001762\n",
      "Written 500 spots for SRR3001762\n",
      "Read 500 spots for SRR3001763\n",
      "Written 500 spots for SRR3001763\n",
      "Read 500 spots for SRR3001764\n",
      "Written 500 spots for SRR3001764\n",
      "Read 500 spots for SRR3001765\n",
      "Written 500 spots for SRR3001765\n",
      "Read 500 spots for SRR3001766\n",
      "Written 500 spots for SRR3001766\n",
      "Read 500 spots for SRR3001767\n",
      "Written 500 spots for SRR3001767\n",
      "Read 500 spots for SRR3001768\n",
      "Written 500 spots for SRR3001768\n",
      "Read 500 spots for SRR3001769\n",
      "Written 500 spots for SRR3001769\n",
      "Read 500 spots for SRR3001770\n",
      "Written 500 spots for SRR3001770\n",
      "Read 500 spots for SRR3001771\n",
      "Written 500 spots for SRR3001771\n",
      "Read 500 spots for SRR3001772\n",
      "Written 500 spots for SRR3001772\n",
      "Read 500 spots for SRR3001773\n",
      "Written 500 spots for SRR3001773\n",
      "Read 500 spots for SRR3001774\n",
      "Written 500 spots for SRR3001774\n",
      "Read 500 spots for SRR3001775\n",
      "Written 500 spots for SRR3001775\n",
      "Read 500 spots for SRR3001776\n",
      "Written 500 spots for SRR3001776\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/rtu/\n",
    "for((a=3001750; a<=3001776 ;a++))\n",
    "do\n",
    "    sratoolkit.2.9.2-ubuntu64/bin/fastq-dump --split-files -X 500 -Z SRR$a > /home/rtu/Sample_fastq/SRR$a.500.fastq\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "2.3) Run _fastqc_ on all the outputs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR3001750.500.fastq\n",
      "Analysis complete for SRR3001751.500.fastq\n",
      "Analysis complete for SRR3001752.500.fastq\n",
      "Analysis complete for SRR3001753.500.fastq\n",
      "Analysis complete for SRR3001754.500.fastq\n",
      "Analysis complete for SRR3001755.500.fastq\n",
      "Analysis complete for SRR3001756.500.fastq\n",
      "Analysis complete for SRR3001757.500.fastq\n",
      "Analysis complete for SRR3001758.500.fastq\n",
      "Analysis complete for SRR3001759.500.fastq\n",
      "Analysis complete for SRR3001760.500.fastq\n",
      "Analysis complete for SRR3001761.500.fastq\n",
      "Analysis complete for SRR3001762.500.fastq\n",
      "Analysis complete for SRR3001763.500.fastq\n",
      "Analysis complete for SRR3001764.500.fastq\n",
      "Analysis complete for SRR3001765.500.fastq\n",
      "Analysis complete for SRR3001766.500.fastq\n",
      "Analysis complete for SRR3001767.500.fastq\n",
      "Analysis complete for SRR3001768.500.fastq\n",
      "Analysis complete for SRR3001769.500.fastq\n",
      "Analysis complete for SRR3001770.500.fastq\n",
      "Analysis complete for SRR3001771.500.fastq\n",
      "Analysis complete for SRR3001772.500.fastq\n",
      "Analysis complete for SRR3001773.500.fastq\n",
      "Analysis complete for SRR3001774.500.fastq\n",
      "Analysis complete for SRR3001775.500.fastq\n",
      "Analysis complete for SRR3001776.500.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of SRR3001750.500.fastq\n",
      "Started analysis of SRR3001751.500.fastq\n",
      "Started analysis of SRR3001752.500.fastq\n",
      "Started analysis of SRR3001753.500.fastq\n",
      "Started analysis of SRR3001754.500.fastq\n",
      "Started analysis of SRR3001755.500.fastq\n",
      "Started analysis of SRR3001756.500.fastq\n",
      "Started analysis of SRR3001757.500.fastq\n",
      "Started analysis of SRR3001758.500.fastq\n",
      "Started analysis of SRR3001759.500.fastq\n",
      "Started analysis of SRR3001760.500.fastq\n",
      "Started analysis of SRR3001761.500.fastq\n",
      "Started analysis of SRR3001762.500.fastq\n",
      "Started analysis of SRR3001763.500.fastq\n",
      "Started analysis of SRR3001764.500.fastq\n",
      "Started analysis of SRR3001765.500.fastq\n",
      "Started analysis of SRR3001766.500.fastq\n",
      "Started analysis of SRR3001767.500.fastq\n",
      "Started analysis of SRR3001768.500.fastq\n",
      "Started analysis of SRR3001769.500.fastq\n",
      "Started analysis of SRR3001770.500.fastq\n",
      "Started analysis of SRR3001771.500.fastq\n",
      "Started analysis of SRR3001772.500.fastq\n",
      "Started analysis of SRR3001773.500.fastq\n",
      "Started analysis of SRR3001774.500.fastq\n",
      "Started analysis of SRR3001775.500.fastq\n",
      "Started analysis of SRR3001776.500.fastq\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/rtu/\n",
    "for i in ./Sample_fastq/*.fastq\n",
    "    do\n",
    "        cd\n",
    "        source activate chip\n",
    "        fastqc $i\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "3.) Run multiqc on the fastqc outputs.What did you find?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[INFO   ]         multiqc : This is MultiQC v1.6 (862256f)\n",
      "[INFO   ]         multiqc : Template    : default\n",
      "[INFO   ]         multiqc : Searching './Sample_fastq/'\n",
      "[INFO   ]          fastqc : Found 27 reports\n",
      "[INFO   ]         multiqc : Compressing plot data\n",
      "[INFO   ]         multiqc : Report      : multiqc_report.html\n",
      "[INFO   ]         multiqc : Data        : multiqc_data\n",
      "[INFO   ]         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/rtu/\n",
    "source activate chip\n",
    "multiqc ./Sample_fastq/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Ran Fastqc and multiqc on each file and recieved and HTML output. Fastqc allows you to analyze one sample at a time, where as multiqc lets you view multiple qc reports at once. \n",
    "\n",
    "My fastqc report on 'SRR3001750.sample.fastq' had a %GC content of 47, sequence length of 50 and 0 flagged as poor quality. Range of quality increased as position in read increased.\n",
    "\n",
    "The multiqc report on 'SRR3001750.500-SRR300174.500'gave an output of the %GC of each fastq run. The range of %GC was as low as 42% to as high as 49%. Lengths of sequences varied. Some were 50, some were 101 bp."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Homework 4.1 ###\n",
    "1. For each file in Sample_fastq:\n",
    "\n",
    "1.1 Use a for loop to trim the fastq files with the minimum read length after trimming 24bp."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.18\n",
      "Analysis complete for SRR3001750.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001751.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001752.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001753.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001754.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001755.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001756.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001757.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001758.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001759.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001760.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001761.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001762.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001763.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001764.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001765.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001766.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001767.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001768.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001769.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001770.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001771.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001772.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001773.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001774.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001775.500_trimmed.fq\n",
      "1.18\n",
      "Analysis complete for SRR3001776.500_trimmed.fq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001750.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count smallRNA: 0, count Nextera: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001750.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001750.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001750.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001750.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001750.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.02 s (34 us/read; 1.74 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       174 (34.8%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,000 bp\n",
      "Quality-trimmed:                   2,242 bp (9.0%)\n",
      "Total written (filtered):         22,514 bp (90.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 174 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.2%\n",
      "  C: 36.2%\n",
      "  G: 13.8%\n",
      "  T: 21.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t122\t125.0\t0\t122\n",
      "2\t37\t31.2\t0\t37\n",
      "3\t12\t7.8\t0\t12\n",
      "4\t3\t2.0\t0\t3\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001750.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t41 (8.2%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001750.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001751.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t17\tAGATCGGAAGAGC\t500\t3.40\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Using Illumina adapter for trimming (count: 17). Second best hit was Nextera (count: 0)\n",
      "\n",
      "Writing report to 'SRR3001751.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001751.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001751.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001751.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001751.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.03 s (56 us/read; 1.08 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       212 (42.4%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        50,500 bp\n",
      "Quality-trimmed:                     576 bp (1.1%)\n",
      "Total written (filtered):         49,197 bp (97.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 212 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 35.4%\n",
      "  C: 26.4%\n",
      "  G: 20.3%\n",
      "  T: 17.9%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t124\t125.0\t0\t124\n",
      "2\t32\t31.2\t0\t32\n",
      "3\t15\t7.8\t0\t15\n",
      "4\t5\t2.0\t0\t5\n",
      "5\t7\t0.5\t0\t7\n",
      "6\t5\t0.1\t0\t5\n",
      "8\t2\t0.0\t0\t2\n",
      "9\t1\t0.0\t0\t1\n",
      "11\t3\t0.0\t1\t3\n",
      "12\t1\t0.0\t1\t1\n",
      "14\t3\t0.0\t1\t3\n",
      "15\t2\t0.0\t1\t2\n",
      "16\t1\t0.0\t1\t1\n",
      "18\t1\t0.0\t1\t1\n",
      "19\t3\t0.0\t1\t3\n",
      "21\t1\t0.0\t1\t1\n",
      "22\t2\t0.0\t1\t2\n",
      "23\t1\t0.0\t1\t1\n",
      "25\t1\t0.0\t1\t1\n",
      "28\t1\t0.0\t1\t1\n",
      "35\t1\t0.0\t1\t1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001751.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t1 (0.2%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001751.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001752.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count Nextera: 0, count Illumina: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001752.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001752.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001752.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001752.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001752.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.03 s (51 us/read; 1.17 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       142 (28.4%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,000 bp\n",
      "Quality-trimmed:                   3,041 bp (12.2%)\n",
      "Total written (filtered):         21,754 bp (87.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 142 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 32.4%\n",
      "  C: 32.4%\n",
      "  G: 19.0%\n",
      "  T: 16.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t100\t125.0\t0\t100\n",
      "2\t30\t31.2\t0\t30\n",
      "3\t9\t7.8\t0\t9\n",
      "4\t2\t2.0\t0\t2\n",
      "10\t1\t0.0\t1\t0 1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001752.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t54 (10.8%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001752.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001753.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count smallRNA: 0, count Nextera: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001753.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001753.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001753.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001753.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001753.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.02 s (41 us/read; 1.48 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       167 (33.4%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        50,500 bp\n",
      "Quality-trimmed:                     360 bp (0.7%)\n",
      "Total written (filtered):         49,908 bp (98.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 167 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.5%\n",
      "  C: 32.9%\n",
      "  G: 24.0%\n",
      "  T: 15.6%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t120\t125.0\t0\t120\n",
      "2\t33\t31.2\t0\t33\n",
      "3\t12\t7.8\t0\t12\n",
      "5\t2\t0.5\t0\t2\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001753.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t0 (0.0%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001753.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001754.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count smallRNA: 0, count Nextera: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001754.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001754.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001754.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001754.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001754.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.02 s (36 us/read; 1.68 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       172 (34.4%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        50,500 bp\n",
      "Quality-trimmed:                     497 bp (1.0%)\n",
      "Total written (filtered):         49,769 bp (98.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 172 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 25.0%\n",
      "  C: 32.6%\n",
      "  G: 20.3%\n",
      "  T: 22.1%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t131\t125.0\t0\t131\n",
      "2\t28\t31.2\t0\t28\n",
      "3\t7\t7.8\t0\t7\n",
      "4\t4\t2.0\t0\t4\n",
      "5\t2\t0.5\t0\t2\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001754.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t0 (0.0%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001754.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001755.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count smallRNA: 0, count Nextera: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001755.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001755.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001755.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001755.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001755.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.03 s (62 us/read; 0.97 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       172 (34.4%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,500 bp\n",
      "Quality-trimmed:                     212 bp (0.8%)\n",
      "Total written (filtered):         25,055 bp (98.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 172 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.5%\n",
      "  C: 28.5%\n",
      "  G: 21.5%\n",
      "  T: 21.5%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t123\t125.0\t0\t123\n",
      "2\t39\t31.2\t0\t39\n",
      "3\t8\t7.8\t0\t8\n",
      "4\t2\t2.0\t0\t2\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001755.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t1 (0.2%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001755.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001756.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count Illumina: 0, count Nextera: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001756.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001756.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001756.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001756.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001756.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.03 s (69 us/read; 0.87 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       181 (36.2%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,000 bp\n",
      "Quality-trimmed:                   2,163 bp (8.7%)\n",
      "Total written (filtered):         22,595 bp (90.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 181 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 26.5%\n",
      "  C: 28.7%\n",
      "  G: 24.3%\n",
      "  T: 20.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t129\t125.0\t0\t129\n",
      "2\t43\t31.2\t0\t43\n",
      "3\t9\t7.8\t0\t9\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001756.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t36 (7.2%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001756.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001757.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count Illumina: 0, count smallRNA: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001757.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001757.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001757.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001757.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001757.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.07 s (143 us/read; 0.42 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       167 (33.4%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        50,500 bp\n",
      "Quality-trimmed:                     221 bp (0.4%)\n",
      "Total written (filtered):         50,062 bp (99.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 167 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.7%\n",
      "  C: 29.9%\n",
      "  G: 21.6%\n",
      "  T: 19.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t131\t125.0\t0\t131\n",
      "2\t28\t31.2\t0\t28\n",
      "3\t7\t7.8\t0\t7\n",
      "9\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001757.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t0 (0.0%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001757.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001758.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t1\tAGATCGGAAGAGC\t500\t0.20\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Using Illumina adapter for trimming (count: 1). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "Writing report to 'SRR3001758.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001758.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001758.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001758.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001758.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.18 s (354 us/read; 0.17 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       184 (36.8%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        50,500 bp\n",
      "Quality-trimmed:                     310 bp (0.6%)\n",
      "Total written (filtered):         49,886 bp (98.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 184 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 25.0%\n",
      "  C: 29.9%\n",
      "  G: 23.9%\n",
      "  T: 21.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t133\t125.0\t0\t133\n",
      "2\t30\t31.2\t0\t30\n",
      "3\t15\t7.8\t0\t15\n",
      "4\t2\t2.0\t0\t2\n",
      "6\t1\t0.1\t0\t1\n",
      "8\t2\t0.0\t0\t2\n",
      "36\t1\t0.0\t1\t1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001758.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t0 (0.0%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001758.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001759.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count Nextera: 0, count smallRNA: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001759.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001759.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001759.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001759.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001759.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.19 s (386 us/read; 0.16 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       165 (33.0%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,000 bp\n",
      "Quality-trimmed:                   2,652 bp (10.6%)\n",
      "Total written (filtered):         22,105 bp (88.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 165 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.5%\n",
      "  C: 33.3%\n",
      "  G: 21.2%\n",
      "  T: 17.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t112\t125.0\t0\t112\n",
      "2\t34\t31.2\t0\t34\n",
      "3\t16\t7.8\t0\t16\n",
      "4\t2\t2.0\t0\t2\n",
      "7\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001759.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t52 (10.4%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001759.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001760.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t11\tAGATCGGAAGAGC\t500\t2.20\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Using Illumina adapter for trimming (count: 11). Second best hit was Nextera (count: 0)\n",
      "\n",
      "Writing report to 'SRR3001760.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001760.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001760.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001760.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001760.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.07 s (132 us/read; 0.46 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       204 (40.8%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        50,500 bp\n",
      "Quality-trimmed:                     408 bp (0.8%)\n",
      "Total written (filtered):         49,579 bp (98.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 204 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.5%\n",
      "  C: 34.3%\n",
      "  G: 18.1%\n",
      "  T: 20.1%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t152\t125.0\t0\t152\n",
      "2\t30\t31.2\t0\t30\n",
      "3\t9\t7.8\t0\t9\n",
      "4\t1\t2.0\t0\t1\n",
      "9\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t1\t1\n",
      "19\t4\t0.0\t1\t4\n",
      "20\t1\t0.0\t1\t1\n",
      "22\t1\t0.0\t1\t1\n",
      "24\t1\t0.0\t1\t1\n",
      "28\t1\t0.0\t1\t1\n",
      "32\t1\t0.0\t1\t1\n",
      "46\t1\t0.0\t1\t1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001760.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t0 (0.0%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001760.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001761.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count smallRNA: 0, count Illumina: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001761.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001761.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001761.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001761.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001761.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.11 s (211 us/read; 0.28 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       166 (33.2%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,000 bp\n",
      "Quality-trimmed:                   2,151 bp (8.6%)\n",
      "Total written (filtered):         22,624 bp (90.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 166 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 30.1%\n",
      "  C: 27.7%\n",
      "  G: 26.5%\n",
      "  T: 15.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t123\t125.0\t0\t123\n",
      "2\t29\t31.2\t0\t29\n",
      "3\t12\t7.8\t0\t12\n",
      "4\t2\t2.0\t0\t2\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001761.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t36 (7.2%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001761.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001762.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t1\tAGATCGGAAGAGC\t500\t0.20\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Using Illumina adapter for trimming (count: 1). Second best hit was smallRNA (count: 0)\n",
      "\n",
      "Writing report to 'SRR3001762.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001762.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001762.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001762.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001762.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.32 s (635 us/read; 0.09 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       183 (36.6%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        50,500 bp\n",
      "Quality-trimmed:                     406 bp (0.8%)\n",
      "Total written (filtered):         49,833 bp (98.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 183 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 33.9%\n",
      "  C: 31.1%\n",
      "  G: 16.4%\n",
      "  T: 18.6%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t131\t125.0\t0\t131\n",
      "2\t33\t31.2\t0\t33\n",
      "3\t17\t7.8\t0\t17\n",
      "4\t1\t2.0\t0\t1\n",
      "9\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001762.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t0 (0.0%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001762.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001763.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count Nextera: 0, count smallRNA: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001763.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001763.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001763.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001763.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001763.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.02 s (45 us/read; 1.33 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       160 (32.0%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        50,500 bp\n",
      "Quality-trimmed:                     571 bp (1.1%)\n",
      "Total written (filtered):         49,697 bp (98.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 160 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 27.5%\n",
      "  C: 32.5%\n",
      "  G: 18.8%\n",
      "  T: 21.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t107\t125.0\t0\t107\n",
      "2\t37\t31.2\t0\t37\n",
      "3\t13\t7.8\t0\t13\n",
      "4\t3\t2.0\t0\t3\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001763.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t2 (0.4%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001763.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001764.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count Nextera: 0, count Illumina: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001764.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001764.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001764.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001764.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001764.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.02 s (43 us/read; 1.39 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       160 (32.0%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,500 bp\n",
      "Quality-trimmed:                     139 bp (0.5%)\n",
      "Total written (filtered):         25,132 bp (98.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 160 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 32.5%\n",
      "  C: 26.9%\n",
      "  G: 23.8%\n",
      "  T: 16.9%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t113\t125.0\t0\t113\n",
      "2\t34\t31.2\t0\t34\n",
      "3\t7\t7.8\t0\t7\n",
      "4\t5\t2.0\t0\t5\n",
      "7\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001764.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t2 (0.4%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001764.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001765.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count smallRNA: 0, count Illumina: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001765.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001765.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001765.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001765.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001765.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.02 s (35 us/read; 1.70 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       193 (38.6%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,000 bp\n",
      "Quality-trimmed:                   2,439 bp (9.8%)\n",
      "Total written (filtered):         22,290 bp (89.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 193 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 29.5%\n",
      "  C: 28.5%\n",
      "  G: 21.2%\n",
      "  T: 20.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t142\t125.0\t0\t142\n",
      "2\t31\t31.2\t0\t31\n",
      "3\t15\t7.8\t0\t15\n",
      "4\t3\t2.0\t0\t3\n",
      "5\t2\t0.5\t0\t2\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001765.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t45 (9.0%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001765.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001766.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count Illumina: 0, count smallRNA: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001766.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001766.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001766.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001766.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001766.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.03 s (61 us/read; 0.99 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       179 (35.8%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        50,500 bp\n",
      "Quality-trimmed:                     304 bp (0.6%)\n",
      "Total written (filtered):         49,963 bp (98.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 179 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 31.8%\n",
      "  C: 27.9%\n",
      "  G: 16.8%\n",
      "  T: 23.5%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t136\t125.0\t0\t136\n",
      "2\t33\t31.2\t0\t33\n",
      "3\t9\t7.8\t0\t9\n",
      "4\t1\t2.0\t0\t1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001766.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t0 (0.0%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001766.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001767.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count smallRNA: 0, count Illumina: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001767.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001767.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001767.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001767.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001767.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.03 s (58 us/read; 1.03 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       178 (35.6%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        50,500 bp\n",
      "Quality-trimmed:                     299 bp (0.6%)\n",
      "Total written (filtered):         49,962 bp (98.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 178 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 30.9%\n",
      "  C: 36.0%\n",
      "  G: 15.7%\n",
      "  T: 17.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t131\t125.0\t0\t131\n",
      "2\t37\t31.2\t0\t37\n",
      "3\t6\t7.8\t0\t6\n",
      "4\t4\t2.0\t0\t4\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001767.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t0 (0.0%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001767.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001768.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count Nextera: 0, count Illumina: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001768.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001768.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001768.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001768.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001768.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.02 s (35 us/read; 1.71 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       176 (35.2%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,500 bp\n",
      "Quality-trimmed:                      87 bp (0.3%)\n",
      "Total written (filtered):         25,175 bp (98.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 176 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.4%\n",
      "  C: 38.1%\n",
      "  G: 19.3%\n",
      "  T: 14.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t136\t125.0\t0\t136\n",
      "2\t24\t31.2\t0\t24\n",
      "3\t11\t7.8\t0\t11\n",
      "4\t4\t2.0\t0\t4\n",
      "5\t1\t0.5\t0\t1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001768.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t0 (0.0%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001768.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001769.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count Illumina: 0, count Nextera: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001769.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001769.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001769.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001769.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001769.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.02 s (42 us/read; 1.43 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       149 (29.8%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,500 bp\n",
      "Quality-trimmed:                     147 bp (0.6%)\n",
      "Total written (filtered):         25,148 bp (98.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 149 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.9%\n",
      "  C: 35.6%\n",
      "  G: 21.5%\n",
      "  T: 14.1%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t105\t125.0\t0\t105\n",
      "2\t35\t31.2\t0\t35\n",
      "3\t7\t7.8\t0\t7\n",
      "4\t1\t2.0\t0\t1\n",
      "5\t1\t0.5\t0\t1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001769.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t0 (0.0%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001769.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001770.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count smallRNA: 0, count Nextera: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001770.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001770.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001770.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001770.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001770.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.02 s (49 us/read; 1.24 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       191 (38.2%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,500 bp\n",
      "Quality-trimmed:                     172 bp (0.7%)\n",
      "Total written (filtered):         25,069 bp (98.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 191 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 31.4%\n",
      "  C: 37.2%\n",
      "  G: 15.7%\n",
      "  T: 15.7%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t136\t125.0\t0\t136\n",
      "2\t44\t31.2\t0\t44\n",
      "3\t9\t7.8\t0\t9\n",
      "4\t2\t2.0\t0\t2\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001770.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t1 (0.2%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001770.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001771.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count Nextera: 0, count smallRNA: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001771.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001771.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001771.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001771.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001771.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.02 s (33 us/read; 1.82 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       149 (29.8%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,500 bp\n",
      "Quality-trimmed:                     201 bp (0.8%)\n",
      "Total written (filtered):         25,034 bp (98.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 149 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 33.6%\n",
      "  C: 30.2%\n",
      "  G: 16.8%\n",
      "  T: 18.8%\n",
      "  none/other: 0.7%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t95\t125.0\t0\t95\n",
      "2\t41\t31.2\t0\t41\n",
      "3\t11\t7.8\t0\t11\n",
      "4\t1\t2.0\t0\t1\n",
      "51\t1\t0.0\t1\t0 1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001771.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t2 (0.4%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001771.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001772.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count smallRNA: 0, count Illumina: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001772.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001772.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001772.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001772.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001772.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.02 s (47 us/read; 1.27 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       177 (35.4%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,500 bp\n",
      "Quality-trimmed:                     133 bp (0.5%)\n",
      "Total written (filtered):         25,061 bp (98.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 177 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 22.6%\n",
      "  C: 40.7%\n",
      "  G: 17.5%\n",
      "  T: 18.6%\n",
      "  none/other: 0.6%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t114\t125.0\t0\t114\n",
      "2\t48\t31.2\t0\t48\n",
      "3\t11\t7.8\t0\t11\n",
      "4\t3\t2.0\t0\t3\n",
      "51\t1\t0.0\t1\t0 1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001772.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t1 (0.2%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001772.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001773.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count smallRNA: 0, count Nextera: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001773.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001773.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001773.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001773.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001773.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.02 s (37 us/read; 1.63 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       168 (33.6%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,500 bp\n",
      "Quality-trimmed:                      81 bp (0.3%)\n",
      "Total written (filtered):         25,189 bp (98.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 168 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 28.0%\n",
      "  C: 32.7%\n",
      "  G: 20.2%\n",
      "  T: 19.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t119\t125.0\t0\t119\n",
      "2\t36\t31.2\t0\t36\n",
      "3\t13\t7.8\t0\t13\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001773.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t0 (0.0%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001773.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001774.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count Illumina: 0, count Nextera: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001774.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001774.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001774.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001774.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001774.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.02 s (38 us/read; 1.58 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       175 (35.0%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,500 bp\n",
      "Quality-trimmed:                     129 bp (0.5%)\n",
      "Total written (filtered):         25,108 bp (98.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 175 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 32.0%\n",
      "  C: 31.4%\n",
      "  G: 24.0%\n",
      "  T: 12.6%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t112\t125.0\t0\t112\n",
      "2\t44\t31.2\t0\t44\n",
      "3\t13\t7.8\t0\t13\n",
      "4\t6\t2.0\t0\t6\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001774.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t0 (0.0%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001774.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001775.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count smallRNA: 0, count Illumina: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001775.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001775.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001775.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001775.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001775.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.03 s (53 us/read; 1.14 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       180 (36.0%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,500 bp\n",
      "Quality-trimmed:                     125 bp (0.5%)\n",
      "Total written (filtered):         25,127 bp (98.5%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 180 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 33.9%\n",
      "  C: 30.0%\n",
      "  G: 20.6%\n",
      "  T: 15.6%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t128\t125.0\t0\t128\n",
      "2\t37\t31.2\t0\t37\n",
      "3\t14\t7.8\t0\t14\n",
      "4\t1\t2.0\t0\t1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001775.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t0 (0.0%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001775.500_trimmed.fq\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> ./SRR3001776.500.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Nextera\t0\tCTGTCTCTTATA\t500\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t500\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t500\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count Nextera: 0, count Illumina: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to 'SRR3001776.500.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: ./SRR3001776.500.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 24 bp\n",
      "Running FastQC on the data once trimming has completed\n",
      "\n",
      "Writing final adapter and quality trimmed output to SRR3001776.500_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file ./SRR3001776.500.fastq <<< \n",
      "This is cutadapt 1.18 with Python 2.7.15\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC ./SRR3001776.500.fastq\n",
      "Processing reads on 1 core in single-end mode ...\n",
      "Finished in 0.02 s (34 us/read; 1.75 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                     500\n",
      "Reads with adapters:                       177 (35.4%)\n",
      "Reads written (passing filters):           500 (100.0%)\n",
      "\n",
      "Total basepairs processed:        25,500 bp\n",
      "Quality-trimmed:                      76 bp (0.3%)\n",
      "Total written (filtered):         25,186 bp (98.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 177 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 29.9%\n",
      "  C: 29.9%\n",
      "  G: 17.5%\n",
      "  T: 22.6%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t134\t125.0\t0\t134\n",
      "2\t30\t31.2\t0\t30\n",
      "3\t9\t7.8\t0\t9\n",
      "4\t3\t2.0\t0\t3\n",
      "5\t1\t0.5\t0\t1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: ./SRR3001776.500.fastq\n",
      "=============================================\n",
      "500 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 24 bp:\t0 (0.0%)\n",
      "\n",
      "\n",
      "  >>> Now running FastQC on the data <<<\n",
      "\n",
      "Started analysis of SRR3001776.500_trimmed.fq\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "source activate chip\n",
    "cd /home/rtu/Sample_fastq\n",
    "\n",
    "for i in ./*.500.fastq\n",
    "do\n",
    "#echo $i\n",
    "    trim_galore --length 24 --fastqc $i\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1.2 Build a new folder named Sample_fastq_trimmed and move all the trimmed fastq files into the new folder."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /home/rtu/Sample_fastq/\n",
    "mkdir Sample_fastq_trimmed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /home/rtu/Sample_fastq/\n",
    "\n",
    "for i in ./*.500_trimmed.fq #moves all the _trimmed.fq files to new folder\n",
    "do\n",
    "    mv $i ./Sample_fastq_trimmed\n",
    "done\n",
    "\n",
    "for i in ./*.500_trimmed_fastqc.html #moves trimmed_fastqc.html to new folder\n",
    "do\n",
    "    mv $i ./Sample_fastq_trimmed\n",
    "done\n",
    "\n",
    "for i in ./*.500_trimmed_fastqc.zip #moves all .zip files\n",
    "do\n",
    "        mv $i ./Sample_fastq_trimmed\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR3001750.500_trimmed_fastqc.html\n",
      "SRR3001750.500_trimmed_fastqc.zip\n",
      "SRR3001750.500_trimmed.fq\n",
      "SRR3001751.500_trimmed_fastqc.html\n",
      "SRR3001751.500_trimmed_fastqc.zip\n",
      "SRR3001751.500_trimmed.fq\n",
      "SRR3001752.500_trimmed_fastqc.html\n",
      "SRR3001752.500_trimmed_fastqc.zip\n",
      "SRR3001752.500_trimmed.fq\n",
      "SRR3001753.500_trimmed_fastqc.html\n",
      "SRR3001753.500_trimmed_fastqc.zip\n",
      "SRR3001753.500_trimmed.fq\n",
      "SRR3001754.500_trimmed_fastqc.html\n",
      "SRR3001754.500_trimmed_fastqc.zip\n",
      "SRR3001754.500_trimmed.fq\n",
      "SRR3001755.500_trimmed_fastqc.html\n",
      "SRR3001755.500_trimmed_fastqc.zip\n",
      "SRR3001755.500_trimmed.fq\n",
      "SRR3001756.500_trimmed_fastqc.html\n",
      "SRR3001756.500_trimmed_fastqc.zip\n",
      "SRR3001756.500_trimmed.fq\n",
      "SRR3001757.500_trimmed_fastqc.html\n",
      "SRR3001757.500_trimmed_fastqc.zip\n",
      "SRR3001757.500_trimmed.fq\n",
      "SRR3001758.500_trimmed_fastqc.html\n",
      "SRR3001758.500_trimmed_fastqc.zip\n",
      "SRR3001758.500_trimmed.fq\n",
      "SRR3001759.500_trimmed_fastqc.html\n",
      "SRR3001759.500_trimmed_fastqc.zip\n",
      "SRR3001759.500_trimmed.fq\n",
      "SRR3001760.500_trimmed_fastqc.html\n",
      "SRR3001760.500_trimmed_fastqc.zip\n",
      "SRR3001760.500_trimmed.fq\n",
      "SRR3001761.500_trimmed_fastqc.html\n",
      "SRR3001761.500_trimmed_fastqc.zip\n",
      "SRR3001761.500_trimmed.fq\n",
      "SRR3001762.500_trimmed_fastqc.html\n",
      "SRR3001762.500_trimmed_fastqc.zip\n",
      "SRR3001762.500_trimmed.fq\n",
      "SRR3001763.500_trimmed_fastqc.html\n",
      "SRR3001763.500_trimmed_fastqc.zip\n",
      "SRR3001763.500_trimmed.fq\n",
      "SRR3001764.500_trimmed_fastqc.html\n",
      "SRR3001764.500_trimmed_fastqc.zip\n",
      "SRR3001764.500_trimmed.fq\n",
      "SRR3001765.500_trimmed_fastqc.html\n",
      "SRR3001765.500_trimmed_fastqc.zip\n",
      "SRR3001765.500_trimmed.fq\n",
      "SRR3001766.500_trimmed_fastqc.html\n",
      "SRR3001766.500_trimmed_fastqc.zip\n",
      "SRR3001766.500_trimmed.fq\n",
      "SRR3001767.500_trimmed_fastqc.html\n",
      "SRR3001767.500_trimmed_fastqc.zip\n",
      "SRR3001767.500_trimmed.fq\n",
      "SRR3001768.500_trimmed_fastqc.html\n",
      "SRR3001768.500_trimmed_fastqc.zip\n",
      "SRR3001768.500_trimmed.fq\n",
      "SRR3001769.500_trimmed_fastqc.html\n",
      "SRR3001769.500_trimmed_fastqc.zip\n",
      "SRR3001769.500_trimmed.fq\n",
      "SRR3001770.500_trimmed_fastqc.html\n",
      "SRR3001770.500_trimmed_fastqc.zip\n",
      "SRR3001770.500_trimmed.fq\n",
      "SRR3001771.500_trimmed_fastqc.html\n",
      "SRR3001771.500_trimmed_fastqc.zip\n",
      "SRR3001771.500_trimmed.fq\n",
      "SRR3001772.500_trimmed_fastqc.html\n",
      "SRR3001772.500_trimmed_fastqc.zip\n",
      "SRR3001772.500_trimmed.fq\n",
      "SRR3001773.500_trimmed_fastqc.html\n",
      "SRR3001773.500_trimmed_fastqc.zip\n",
      "SRR3001773.500_trimmed.fq\n",
      "SRR3001774.500_trimmed_fastqc.html\n",
      "SRR3001774.500_trimmed_fastqc.zip\n",
      "SRR3001774.500_trimmed.fq\n",
      "SRR3001775.500_trimmed_fastqc.html\n",
      "SRR3001775.500_trimmed_fastqc.zip\n",
      "SRR3001775.500_trimmed.fq\n",
      "SRR3001776.500_trimmed_fastqc.html\n",
      "SRR3001776.500_trimmed_fastqc.zip\n",
      "SRR3001776.500_trimmed.fq\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/rtu/Sample_fastq/Sample_fastq_trimmed\n",
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1.3 Run fastqc for all the trimmed files and compile the results with multiqc. summarize in 1 sentence the difference between multiqc output of trimmed and untrimmed files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for SRR3001750.500.fastq\n",
      "Analysis complete for SRR3001750.sample.fastq\n",
      "Analysis complete for SRR3001751.500.fastq\n",
      "Analysis complete for SRR3001752.500.fastq\n",
      "Analysis complete for SRR3001753.500.fastq\n",
      "Analysis complete for SRR3001754.500.fastq\n",
      "Analysis complete for SRR3001755.500.fastq\n",
      "Analysis complete for SRR3001756.500.fastq\n",
      "Analysis complete for SRR3001757.500.fastq\n",
      "Analysis complete for SRR3001758.500.fastq\n",
      "Analysis complete for SRR3001759.500.fastq\n",
      "Analysis complete for SRR3001760.500.fastq\n",
      "Analysis complete for SRR3001761.500.fastq\n",
      "Analysis complete for SRR3001762.500.fastq\n",
      "Analysis complete for SRR3001763.500.fastq\n",
      "Analysis complete for SRR3001764.500.fastq\n",
      "Analysis complete for SRR3001765.500.fastq\n",
      "Analysis complete for SRR3001766.500.fastq\n",
      "Analysis complete for SRR3001767.500.fastq\n",
      "Analysis complete for SRR3001768.500.fastq\n",
      "Analysis complete for SRR3001769.500.fastq\n",
      "Analysis complete for SRR3001770.500.fastq\n",
      "Analysis complete for SRR3001771.500.fastq\n",
      "Analysis complete for SRR3001772.500.fastq\n",
      "Analysis complete for SRR3001773.500.fastq\n",
      "Analysis complete for SRR3001774.500.fastq\n",
      "Analysis complete for SRR3001775.500.fastq\n",
      "Analysis complete for SRR3001776.500.fastq\n",
      "Analysis complete for SRR5830636.100.fastq\n",
      "Analysis complete for SRR5830637.100.fastq\n",
      "Analysis complete for SRR5830638.100.fastq\n",
      "Analysis complete for SRR5830639.100.fastq\n",
      "Analysis complete for SRR5830640.100.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of SRR3001750.500.fastq\n",
      "Started analysis of SRR3001750.sample.fastq\n",
      "Approx 5% complete for SRR3001750.sample.fastq\n",
      "Approx 10% complete for SRR3001750.sample.fastq\n",
      "Approx 15% complete for SRR3001750.sample.fastq\n",
      "Approx 20% complete for SRR3001750.sample.fastq\n",
      "Approx 25% complete for SRR3001750.sample.fastq\n",
      "Approx 30% complete for SRR3001750.sample.fastq\n",
      "Approx 35% complete for SRR3001750.sample.fastq\n",
      "Approx 40% complete for SRR3001750.sample.fastq\n",
      "Approx 45% complete for SRR3001750.sample.fastq\n",
      "Approx 50% complete for SRR3001750.sample.fastq\n",
      "Approx 55% complete for SRR3001750.sample.fastq\n",
      "Approx 60% complete for SRR3001750.sample.fastq\n",
      "Approx 65% complete for SRR3001750.sample.fastq\n",
      "Approx 70% complete for SRR3001750.sample.fastq\n",
      "Approx 75% complete for SRR3001750.sample.fastq\n",
      "Approx 80% complete for SRR3001750.sample.fastq\n",
      "Approx 85% complete for SRR3001750.sample.fastq\n",
      "Approx 90% complete for SRR3001750.sample.fastq\n",
      "Approx 95% complete for SRR3001750.sample.fastq\n",
      "Approx 100% complete for SRR3001750.sample.fastq\n",
      "Started analysis of SRR3001751.500.fastq\n",
      "Started analysis of SRR3001752.500.fastq\n",
      "Started analysis of SRR3001753.500.fastq\n",
      "Started analysis of SRR3001754.500.fastq\n",
      "Started analysis of SRR3001755.500.fastq\n",
      "Started analysis of SRR3001756.500.fastq\n",
      "Started analysis of SRR3001757.500.fastq\n",
      "Started analysis of SRR3001758.500.fastq\n",
      "Started analysis of SRR3001759.500.fastq\n",
      "Started analysis of SRR3001760.500.fastq\n",
      "Started analysis of SRR3001761.500.fastq\n",
      "Started analysis of SRR3001762.500.fastq\n",
      "Started analysis of SRR3001763.500.fastq\n",
      "Started analysis of SRR3001764.500.fastq\n",
      "Started analysis of SRR3001765.500.fastq\n",
      "Started analysis of SRR3001766.500.fastq\n",
      "Started analysis of SRR3001767.500.fastq\n",
      "Started analysis of SRR3001768.500.fastq\n",
      "Started analysis of SRR3001769.500.fastq\n",
      "Started analysis of SRR3001770.500.fastq\n",
      "Started analysis of SRR3001771.500.fastq\n",
      "Started analysis of SRR3001772.500.fastq\n",
      "Started analysis of SRR3001773.500.fastq\n",
      "Started analysis of SRR3001774.500.fastq\n",
      "Started analysis of SRR3001775.500.fastq\n",
      "Started analysis of SRR3001776.500.fastq\n",
      "Started analysis of SRR5830636.100.fastq\n",
      "Started analysis of SRR5830637.100.fastq\n",
      "Started analysis of SRR5830638.100.fastq\n",
      "Started analysis of SRR5830639.100.fastq\n",
      "Started analysis of SRR5830640.100.fastq\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/rtu/ #untrimmed files\n",
    "for i in ./Sample_fastq/*.fastq\n",
    "    do\n",
    "        cd\n",
    "        source activate chip\n",
    "        fastqc $i\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[INFO   ]         multiqc : This is MultiQC v1.6 (862256f)\n",
      "[INFO   ]         multiqc : Template    : default\n",
      "[INFO   ]         multiqc : Searching './Sample_fastq_trimmed/'\n",
      "[INFO   ]          fastqc : Found 27 reports\n",
      "[INFO   ]         multiqc : Compressing plot data\n",
      "[WARNING]         multiqc : Previous MultiQC output found! Adjusting filenames..\n",
      "[WARNING]         multiqc : Use -f or --force to overwrite existing reports instead\n",
      "[INFO   ]         multiqc : Report      : multiqc_report_1.html\n",
      "[INFO   ]         multiqc : Data        : multiqc_data_1\n",
      "[INFO   ]         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/rtu/\n",
    "source activate chip\n",
    "multiqc ./Sample_fastq_trimmed/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[INFO   ]         multiqc : This is MultiQC v1.6 (862256f)\n",
      "[INFO   ]         multiqc : Template    : default\n",
      "[INFO   ]         multiqc : Searching './Sample_fastq/'\n",
      "[INFO   ]        cutadapt : Found 27 reports\n",
      "[INFO   ]          fastqc : Found 27 reports\n",
      "[INFO   ]         multiqc : Compressing plot data\n",
      "[WARNING]         multiqc : Previous MultiQC output found! Adjusting filenames..\n",
      "[WARNING]         multiqc : Use -f or --force to overwrite existing reports instead\n",
      "[INFO   ]         multiqc : Report      : multiqc_report_2.html\n",
      "[INFO   ]         multiqc : Data        : multiqc_data_2\n",
      "[INFO   ]         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/rtu/\n",
    "source activate chip\n",
    "multiqc ./Sample_fastq/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Differences between trimmed an untrimmed report.\n",
    "\n",
    "-Untrimmed report has a section on Cutadapt, which is a tool for removing adapter sequences, primers and poly A tails.\n",
    "\n",
    "-Trimmed report has uneven number of reads. Untrimmed has even amount of reads. This makes sense.\n",
    "\n",
    "-In Sequence Quality Histogram: Trimmed report has all 27 samples having a mean Phred Score >30. Untrimmed has 2 under 30.\n",
    "\n",
    "-There is more variation in untrimmed multiqc report, with length distribution around ~50 and ~100. Trimmed is mainly around ~100."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "2. Alignment\n",
    "\n",
    "2.1 Build bowtie2 index with genome_chr12.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Settings:\n",
      "  Output files: \"genome_chr12.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  genome_chr12.fa\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:01\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:01\n",
      "bmax according to bmaxDivN setting: 29230605\n",
      "Using parameters --bmax 21922954 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 21922954 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:02\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 1.16922e+08 (target: 21922953)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 116922420 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:43\n",
      "Returning block of 116922421 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 33897029\n",
      "fchr[G]: 58271369\n",
      "fchr[T]: 82696781\n",
      "fchr[$]: 116922420\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 43168663 bytes to primary EBWT file: genome_chr12.1.bt2\n",
      "Wrote 29230612 bytes to secondary EBWT file: genome_chr12.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 116922420\n",
      "    bwtLen: 116922421\n",
      "    sz: 29230605\n",
      "    bwtSz: 29230606\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 7307652\n",
      "    offsSz: 29230608\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 608971\n",
      "    numLines: 608971\n",
      "    ebwtTotLen: 38974144\n",
      "    ebwtTotSz: 38974144\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:01:01\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:01\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "  Time to reverse reference sequence: 00:00:00\n",
      "bmax according to bmaxDivN setting: 29230605\n",
      "Using parameters --bmax 21922954 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 21922954 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:02\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 1.16922e+08 (target: 21922953)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 116922420 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:41\n",
      "Returning block of 116922421 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 33897029\n",
      "fchr[G]: 58271369\n",
      "fchr[T]: 82696781\n",
      "fchr[$]: 116922420\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 43168663 bytes to primary EBWT file: genome_chr12.rev.1.bt2\n",
      "Wrote 29230612 bytes to secondary EBWT file: genome_chr12.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 116922420\n",
      "    bwtLen: 116922421\n",
      "    sz: 29230605\n",
      "    bwtSz: 29230606\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 7307652\n",
      "    offsSz: 29230608\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 608971\n",
      "    numLines: 608971\n",
      "    ebwtTotLen: 38974144\n",
      "    ebwtTotSz: 38974144\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:00:59\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Building a SMALL index\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/rtu/\n",
    "source activate chip\n",
    "bowtie2-build -f genome_chr12.fa genome_chr12"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "2.2 For files in Sample_fastq_trimmed, align all to chr12 into sam files, and compile all the stderr to a file named all.align.out."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /home/rtu/\n",
    "source activate chip\n",
    "abowtieindex=/home/rtu/genome_chr12\n",
    "\n",
    "for i in ./Sample_fastq_data/Sample_fastq_trimmed/*.fq\n",
    "do\n",
    "bowtie2 -x genome_chr12 -U $i -S $i.sample.nontrimmed.sam &> $i.sample.all.align.out #check this file? check export path. source acativate chip?\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "2.3 Filter the all.align.out file to find out for each fastqfile, what percentage of reads aligned exactly 1 time?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    29 (6.32%) aligned exactly 1 time\n",
      "    53 (10.62%) aligned exactly 1 time\n",
      "    55 (12.33%) aligned exactly 1 time\n",
      "    51 (10.20%) aligned exactly 1 time\n",
      "    55 (11.00%) aligned exactly 1 time\n",
      "    30 (6.01%) aligned exactly 1 time\n",
      "    26 (5.60%) aligned exactly 1 time\n",
      "    72 (14.40%) aligned exactly 1 time\n",
      "    66 (13.20%) aligned exactly 1 time\n",
      "    32 (7.14%) aligned exactly 1 time\n",
      "    39 (7.80%) aligned exactly 1 time\n",
      "    70 (15.09%) aligned exactly 1 time\n",
      "    62 (12.40%) aligned exactly 1 time\n",
      "    55 (11.04%) aligned exactly 1 time\n",
      "    39 (7.83%) aligned exactly 1 time\n",
      "    41 (9.01%) aligned exactly 1 time\n",
      "    76 (15.20%) aligned exactly 1 time\n",
      "    91 (18.20%) aligned exactly 1 time\n",
      "    54 (10.80%) aligned exactly 1 time\n",
      "    65 (13.00%) aligned exactly 1 time\n",
      "    47 (9.42%) aligned exactly 1 time\n",
      "    35 (7.03%) aligned exactly 1 time\n",
      "    26 (5.21%) aligned exactly 1 time\n",
      "    50 (10.00%) aligned exactly 1 time\n",
      "    62 (12.40%) aligned exactly 1 time\n",
      "    47 (9.40%) aligned exactly 1 time\n",
      "    57 (11.40%) aligned exactly 1 time\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/rtu/Sample_fastq_data/Sample_fastq_trimmed\n",
    "\n",
    "for i in ./*.align.out #test for loop. it works.\n",
    "do\n",
    "    awk 'NR==4{print}' $i\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "3. For all the alignment output sam files:\n",
    "\n",
    "3.1 convert to bam, sort and filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bam_rmdupse_core] 0 / 68 = 0.0000 in library '\t'\n",
      "[bam_rmdupse_core] 3 / 97 = 0.0309 in library '\t'\n",
      "[bam_rmdupse_core] 3 / 97 = 0.0309 in library '\t'\n",
      "[bam_rmdupse_core] 1 / 100 = 0.0100 in library '\t'\n",
      "[bam_rmdupse_core] 0 / 95 = 0.0000 in library '\t'\n",
      "[bam_rmdupse_core] 0 / 90 = 0.0000 in library '\t'\n",
      "[bam_rmdupse_core] 0 / 94 = 0.0000 in library '\t'\n",
      "[bam_rmdupse_core] 6 / 121 = 0.0496 in library '\t'\n",
      "[bam_rmdupse_core] 3 / 116 = 0.0259 in library '\t'\n",
      "[bam_rmdupse_core] 0 / 80 = 0.0000 in library '\t'\n",
      "[bam_rmdupse_core] 1 / 83 = 0.0120 in library '\t'\n",
      "[bam_rmdupse_core] 6 / 115 = 0.0522 in library '\t'\n",
      "[bam_rmdupse_core] 5 / 114 = 0.0439 in library '\t'\n",
      "[bam_rmdupse_core] 1 / 103 = 0.0097 in library '\t'\n",
      "[bam_rmdupse_core] 1 / 82 = 0.0122 in library '\t'\n",
      "[bam_rmdupse_core] 1 / 99 = 0.0101 in library '\t'\n",
      "[bam_rmdupse_core] 7 / 125 = 0.0560 in library '\t'\n",
      "[bam_rmdupse_core] 8 / 147 = 0.0544 in library '\t'\n",
      "[bam_rmdupse_core] 2 / 113 = 0.0177 in library '\t'\n",
      "[bam_rmdupse_core] 5 / 135 = 0.0370 in library '\t'\n",
      "[bam_rmdupse_core] 0 / 96 = 0.0000 in library '\t'\n",
      "[bam_rmdupse_core] 0 / 83 = 0.0000 in library '\t'\n",
      "[bam_rmdupse_core] 0 / 83 = 0.0000 in library '\t'\n",
      "[bam_rmdupse_core] 0 / 124 = 0.0000 in library '\t'\n",
      "[bam_rmdupse_core] 6 / 148 = 0.0405 in library '\t'\n",
      "[bam_rmdupse_core] 1 / 116 = 0.0086 in library '\t'\n",
      "[bam_rmdupse_core] 3 / 139 = 0.0216 in library '\t'\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/rtu/Sample_fastq_data/Sample_fastq_trimmed\n",
    "\n",
    "for i in *.sam #test for loop. it works.\n",
    "do\n",
    "samtools view -bS $i > ${i:0:10}.bam #$i from indexes 0:10 is the SRR file name\n",
    "samtools sort ${i:0:10}.bam -o ${i:0:10}.srt.bam\n",
    "samtools rmdup -S ${i:0:10}.srt.bam ${i:0:10}.srt.dupr.bam\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "3.2 Summarize the sizes for all the sam files/bam files/ sorted bam files/ sorted and filtered bam files into a txt file named \"sum_sizes.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "160K SRR3001750.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001750.bam\n",
      " 48K SRR3001750.srt.bam\n",
      " 48K SRR3001750.srt.dupr.bam\n",
      "160K SRR3001751.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 56K SRR3001751.bam\n",
      " 48K SRR3001751.srt.bam\n",
      " 48K SRR3001751.srt.dupr.bam\n",
      "160K SRR3001752.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001752.bam\n",
      " 48K SRR3001752.srt.bam\n",
      " 48K SRR3001752.srt.dupr.bam\n",
      "160K SRR3001753.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001753.bam\n",
      " 48K SRR3001753.srt.bam\n",
      " 48K SRR3001753.srt.dupr.bam\n",
      "160K SRR3001754.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001754.bam\n",
      " 48K SRR3001754.srt.bam\n",
      " 48K SRR3001754.srt.dupr.bam\n",
      "160K SRR3001755.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001755.bam\n",
      " 48K SRR3001755.srt.bam\n",
      " 48K SRR3001755.srt.dupr.bam\n",
      "160K SRR3001756.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 56K SRR3001756.bam\n",
      " 48K SRR3001756.srt.bam\n",
      " 48K SRR3001756.srt.dupr.bam\n",
      "160K SRR3001757.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001757.bam\n",
      " 48K SRR3001757.srt.bam\n",
      " 48K SRR3001757.srt.dupr.bam\n",
      "160K SRR3001758.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001758.bam\n",
      " 48K SRR3001758.srt.bam\n",
      " 48K SRR3001758.srt.dupr.bam\n",
      "160K SRR3001759.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001759.bam\n",
      " 48K SRR3001759.srt.bam\n",
      " 48K SRR3001759.srt.dupr.bam\n",
      "160K SRR3001760.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 56K SRR3001760.bam\n",
      " 56K SRR3001760.srt.bam\n",
      " 56K SRR3001760.srt.dupr.bam\n",
      "160K SRR3001761.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001761.bam\n",
      " 48K SRR3001761.srt.bam\n",
      " 48K SRR3001761.srt.dupr.bam\n",
      "160K SRR3001762.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001762.bam\n",
      " 48K SRR3001762.srt.bam\n",
      " 48K SRR3001762.srt.dupr.bam\n",
      "160K SRR3001763.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001763.bam\n",
      " 48K SRR3001763.srt.bam\n",
      " 48K SRR3001763.srt.dupr.bam\n",
      "160K SRR3001764.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001764.bam\n",
      " 48K SRR3001764.srt.bam\n",
      " 48K SRR3001764.srt.dupr.bam\n",
      "160K SRR3001765.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001765.bam\n",
      " 48K SRR3001765.srt.bam\n",
      " 48K SRR3001765.srt.dupr.bam\n",
      "160K SRR3001766.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001766.bam\n",
      " 48K SRR3001766.srt.bam\n",
      " 48K SRR3001766.srt.dupr.bam\n",
      "160K SRR3001767.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001767.bam\n",
      " 48K SRR3001767.srt.bam\n",
      " 48K SRR3001767.srt.dupr.bam\n",
      "160K SRR3001768.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 40K SRR3001768.bam\n",
      " 40K SRR3001768.srt.bam\n",
      " 40K SRR3001768.srt.dupr.bam\n",
      "160K SRR3001769.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001769.bam\n",
      " 40K SRR3001769.srt.bam\n",
      " 40K SRR3001769.srt.dupr.bam\n",
      "160K SRR3001770.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001770.bam\n",
      " 40K SRR3001770.srt.bam\n",
      " 40K SRR3001770.srt.dupr.bam\n",
      "160K SRR3001771.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001771.bam\n",
      " 40K SRR3001771.srt.bam\n",
      " 40K SRR3001771.srt.dupr.bam\n",
      "160K SRR3001772.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001772.bam\n",
      " 48K SRR3001772.srt.bam\n",
      " 48K SRR3001772.srt.dupr.bam\n",
      "160K SRR3001773.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001773.bam\n",
      " 48K SRR3001773.srt.bam\n",
      " 48K SRR3001773.srt.dupr.bam\n",
      "160K SRR3001774.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001774.bam\n",
      " 40K SRR3001774.srt.bam\n",
      " 40K SRR3001774.srt.dupr.bam\n",
      "256K SRR3001775.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 48K SRR3001775.bam\n",
      " 40K SRR3001775.srt.bam\n",
      " 40K SRR3001775.srt.dupr.bam\n",
      "160K SRR3001776.500_trimmed.fq.sample.nontrimmed.sam\n",
      " 40K SRR3001776.bam\n",
      " 40K SRR3001776.srt.bam\n",
      " 40K SRR3001776.srt.dupr.bam\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/rtu/Sample_fastq_data/Sample_fastq_trimmed/\n",
    "ls -sh {*.sam,*.bam*} > sum_sizes.txt\n",
    "less sum_sizes.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
